lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -34,8 +35,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
     }
 
     // Use custom serialize if it exists.
-    if (handler.serialize) {
-      return handler.serialize(model, names);
+    if (serializer) {
+      return serializer(model, names);
```

Comment:
Mine is not a use case that I'd use to argue for "fixing" this bug. For complicated reasons (that I would have probably avoided had I been as versed in Ember as I am now a year a half ago when I wrote it), my afterModel hook pushes some state into a service, and my serialize hook needs to recover it, so it does so via an injection on the route object. I suppose I could ```getOwner(model).lookup('service:...')``` rather than use an injection, but I think the whole setup that I have is kind of misguided. Anyway, I solved it by

```javascript
init() {
  this._super(...arguments);
  this.serialize = (model) => {
    // ...
  };
}
```

I agree with making the serializer a pure transform, because that would drive cleaner patterns rather than the mess I've gotten myself into. However, that being the case I think it's very counter-intuitive to have the serializer as a property of the router, because then it looks like an instance method hook like the other model hooks, and my expectation is that it is called with ```this``` bound to the route. But I guess you guys are already talking about deprecating it in favor of putting the serializer in the route map? Makes sense to me.

But now while it's still available as what looks like a route hook, I don't think it really makes sense to have it unbound. But, as you point out, I have a workaround and since that likely represents 100% of the people encountering this problem, I can see it making sense to just leave it since this form of serialize is going away anyway.